### PR TITLE
Bloodview improvements

### PR DESCRIPTION
**Pull request by tlsa**
At a high level, this:

* Allows graphs to have their own colours.
* Adds support for flash patterns.
* Changes the main menu to be defined by a YAML file.
* Adds default configuration files for revision 1 and 2 hardware.

It also includes a small change to the firmware to fix flash pattern acquisitions.

Note that this PR adds a dep on [LibCYAML](https://github.com/tlsa/libcyaml), which is added as a submodule.

* Closes: #61
* Closes: #56
* Closes: #54
* Closes: #37

### CODE REVIEW

**Comment by jonathanmaw** [comment]
The changes work for me, and the code looks fine to me but I haven't touched sdl before

**Comment by tlsa** [comment]
> The changes work for me, and the code looks fine to me but I haven't touched sdl before

Cool, thanks for testing!

**Comment by tlsa** [comment]
Rebased on top of merged #59.

File: bloodview/src/device.c [hunk 1]
```diff
@@ -412,6 +414,10 @@ static void *device__thread(void *ctx)
 			/* Awaiting response to sent message,
 			 * or running an acquisition. */
 			device__thread_receive_msg(&sent_type);
+		} else {
+			/* When we're not waiting for messages, don't
+			 * thrash the device thread main loop. */
+			usleep(10 * 1000);
```
Comment (ben-brewer-codethink) [inline bloodview/src/device.c:449]: It think you can use sleep(0) to concede timeslice without needing to give a timeout time for situations like this, not sure if useful though.

**Review by ben-brewer-codethink — APPROVED** [review]
Looks good to me.

File: bloodview/src/device.c (hunk 1 above)
Comment (tlsa) [reply bloodview/src/device.c:449]: OK, cool.  I'll leave it as 10ms for now, because we don't really need frequent polling here.

**Pull request merged**